lib/engines/object_store/docs/adr/0001-create-object-store.md

```diff
@@ -0,0 +1,56 @@
+# 3. Build an Object Store in Whitehall to create/edit Content Blocks 
+
+Date: 2024-06-20
+
+## Context
+
+The content modelling team is going to be building features to allow small reusable content objects to be included or 
+embedded into other GOV.UK documents. We plan to allow consistent reuse, and edit-once update-everywhere behaviour, valuable 
+during uprating for example, or when highly-duplicated content, like phone numbers and email addresses change.
+
+We want to ensure the following:
+
+* What we build should in theory be able to be used by any publishing app
+* Content Blocks have the same publishing process as documents (drafts, 2i, authorisation etc)
+* When a Content Block is updated, any documents referencing it should also be updated
+* Content Blocks can be categorised with meaning (so they're not just blocks of text)
+* The structure of user-facing content within Content Blocks is customisable (Examples of content are email addresses and 
+tax rates, but could include much more complicated structured data. )
+
+
+## Decisions and Consequences
+
+Our first step towards building this service is to begin building the create/edit functionality.
+
+### 1. We will build the internal user-facing service in Whitehall
+
+All our users who may need to add/edit content blocks should have access to the Whitehall app. 
+
+### 2. We will build it in an engine
+
+We want to ensure that we are 'good neighbours' and can create this service in a way that isn't polluting the Whitehall
+repo. We also want to build with a view to this service being used outside of Whitehall. It makes sense therefore to put 
+the code in an engine separate from the rest of the repo.
+
+### 3. We will make Edition-like behaviour into a concern
+
+We think that Content Blocks will need some behaviour that belongs to Editions, such as being related to a Document and 
+having versions/2i. But we are aware that inheriting from [Editions may cause dependency issues](https://github.com/alphagov/whitehall/pull/9180#discussion_r1650538039) as it would mean integrating with an already 
+very complex and wide-ranging model and system. 
+We will plan to add a new Editionable Concern that's private to us only that (for now):
+
+* Specifies editions as the table we want to use (with self.table_name)
+* Includes the Edition::Identifiable concern (which sets up the relation between the Editions and the Document)
+
+As we progress we can then identify the behaviours we need and import them into our Edtionable concern. This will also 
+have the side effect of separating out the stuff that makes our model editionable from the stuff that is specific to it 
+being a content block (i.e. the JSON columns, validation etc) and will also help us identify the bare minimum of behaviour 
+required to make a model behave like an edition.
+
+### 4. The source of truth for published Blocks will be the Publishing API
```

Comment:
ah ok, I copied this [over from the original ADR](https://docs.google.com/document/d/1b-jgRmr-CrtuUhPkgPZbFBt1UG0f8wJ4hOFLTAS00D0/edit)
I assumed this meant that for any other services wanting to get what the latest Content Block is they will talk to the Publishing API rather than Whitehall... but Whitehall will be pushing out the latest versions and keeping track of the blocks. Maybe there's a better of way of wording this

Reply:
Sounds like Whitehall is the source of truth here and we are synchronising _published_ blocks with the publishing API via a push?

Then when the publishing API has to translate some references it can either check locally in it's copy or ask Whitehall.

If that's correct and Publishing API just has a secondary copy that's only for reading.. I wonder if a cache with a fallback request to Whitehall wouldn't be lighter and simpler?

Reply:
We send all versions (drafts and published versions) to the content API. Drafts get sent to the draft content store (for previewing etc), while published versions get sent to the live content store. I did wonder about the wisdom of keeping a copy in the database too, but things like publishing flow and who created an edition and when isn't information that the Publishing API keeps. We'd also have to carry out validations and search etc via the API, which may result in slowness or missing features that we'd have to implement on the publishing API.